### ISSUE TITLE
Add option to rename tag/field names based on regex

### ISSUE THREAD

**Issue by Hipska** [influxdata/telegraf#9506]
## Feature Request

Add options to also rename the tag and field names instead of only just the values.
The regex processor seems to be ideal to add this functionality to.

### Proposal:

```toml
[[processors.regex]]
  namepass = ["snmp_trap"]

  # Tag and field conversions defined in a separate sub-tables
  [[processors.regex.tag_rename]]
    ## Regular expression to match on a tag name
    pattern = '^(\w+)\.\d+$'
    ## Matches of the pattern will be replaced with this string.  Use ${1}
    ## notation to use the text of the first submatch.
    replacement = "${1}"

  [[processors.regex.field_rename]]
    ## Regular expression to match on a field name
    pattern = '^(\w+)\.\d+$'
    ## Matches of the pattern will be replaced with this string.  Use ${1}
    ## notation to use the text of the first submatch.
    replacement = "${1}"
```

### Current behavior:

_Not implemented._

### Desired behavior:

```diff
- snmp_trap,mib=IF-MIB,name=linkDown,oid=.1.3.6.1.6.3.1.1.5.3,source=192.168.0.1,version=2c,community=public,ifIndex.3=3i,ifAdminStatus.3=1i sysUpTimeInstance=5803i,ifOperStatus.3=2i 1574109186555115459
+ snmp_trap,mib=IF-MIB,name=linkDown,oid=.1.3.6.1.6.3.1.1.5.3,source=192.168.0.1,version=2c,community=public,ifIndex=3i,ifAdminStatus=1i sysUpTimeInstance=5803i,ifOperStatus=2i 1574109186555115459
```

### Use case:

SNMP traps are usually sent with indexed fields, so it is very difficult to process them later.

**Comment by srebhan** [comment]
Related to #5159.

**Comment by srebhan** [comment]
@Hipska can you please check if #9561 works for your case? If you encounter any problem, please let me know.

**Comment by Hipska** [comment]
@srebhan Yes, that is indeed a way to get the desired behaviour.

**Comment by MyaLongmire** [comment]
Since this works as intended I am going to close the issue.

**Comment by Hipska** [comment]
Yeah, but it’s not merged yet. So issue should stay open?

**Comment by srebhan** [comment]
@MyaLongmire the issue will be closed automatically in case you merge #9561. Otherwise, there is no way to solve the problem described in this issue.
